### DISCUSSION TITLE
In the HTTP protocol, the 'Connection: keep-alive' is ineffective

### DISCUSSION THREAD

**Discussion by AlexOct** [cesanta/mongoose#2833] [Ask questions, get answers] [answered]
I requested a WebSocket and HTTP in the frontend, both sharing port 8000. I want HTTP to use a persistent connection for image transmission. However, even though I used 'Connection: keep-alive', the TCP connection still disconnects and establishes a new connection (observed through TCP port numbers).

**Comment by scaprile** [comment, marked as answer]
I don't know what you are doing.
I don't know what you mean by "frontend", and since it is a frontend and establishes a connection, is a client, it can't "share" port 8000.
Please see our [documentation](https://mongoose.ws/documentation/),  and follow the guidelines in our [tutorials](https://mongoose.ws/documentation/#tutorials-and-examples).
If there is an actual issue, please open an issue, _honor the issue template_ so we can know your scenario and reproduce the issue if necessary, and let us know _exactly_ what you are doing

**Reply by AlexOct** [reply]
Thank you for reply. Sorry for my unclear description. I'm a beginner. I will review it more.

**Comment by pseudotronics** [comment]
@AlexOct YSK that modern web browsers will timeout websocket connections that sit idle. The solution is to implement a heartbeat or ping-pong message to keep the connection alive.
